HealthcareSystem/GraphicalEditor/Repository/FileRepository.cs

```diff
@@ -27,6 +27,7 @@ public void SaveMap(List<MapObject> allMapObjects)
             serializer.TypeNameHandling = TypeNameHandling.All;
             serializer.Formatting = Formatting.Indented;
             serializer.ContractResolver = new ContractResolver();
+            serializer.ReferenceLoopHandling = ReferenceLoopHandling.Ignore;
```

Comment:
Also, I assume that some of the changes here were meant to be some sort of fix for the serialization & deserialization of the map? Not sure if you know but that still doesn't work (at least when I try it, deserialization fails after the map is serialized), so I suggest making a fix branch for that sometime in the future.

Reply:
The suggestion you mentioned for the EditModeActive flag is definitely approach I was planning to implement. I've had small problems with that approach and my colleague wanted to start implementing edit functionality, so I left it with this. Refactoring will probably be neccessary when we start the next sprint. Making two stack panels is a good suggestion. I will definetely consider it.
Editing won't look like that. Changes to a field won't be lost by focusing on another field. Only way to discard the changes is by clicking the Cancel button that will be implemented.
As serialization is concerned, I guess the problem is that you do not have a map saved. Try saving map objects from MockupObjects class, commenting that line and then load from file.

Reply:
So I tried checking the deserialization issue again, and it's definitely not because there's no map saved, it happens specifically if I create the mockup objects, save them, and then load. It gets kind of odd though, because on this branch I get an exception no matter how I run, but on another branch (feature-g8.1-all-objectinfo-display) it doesn't happen if I run the editor directly, but does if I run from the ManagerFrontend. Here's the exception I'm getting:
![image](https://user-images.githubusercontent.com/57844740/98418556-70c9a580-2083-11eb-9d13-4162251aed64.png)
And the call stack:
![image](https://user-images.githubusercontent.com/57844740/98418617-9060ce00-2083-11eb-83a7-4fc1116a6efb.png)

Could just be something wrong on my copy of the repository, but maybe try reproducing the issue?

Reply:
I am not quite sure what is going on here. I haven't been able to reproduce that exception. This plugin has been producing a lot of weird bugs since I started using it. It seems like I will need to write saving and retrieving from 0 like I initially planned, but the problem is that model is constantly evolving, so I will need to change that class constantly...